src/mpyl/steps/collection.py

```diff
@@ -11,9 +12,25 @@
 class StepsCollection:
     _step_executors: set[Step]
 
-    def __init__(self, logger: Logger, base_path: Optional[str] = None) -> None:
+    def __init__(self, logger: Logger) -> None:
         self._step_executors = set()
-        self.__load_steps_in_module(".", base_path)
+        package = "mpyl.steps"
+        local_package = f"src.{package}"
+        default_location, alternative_location = (
+            (local_package, package)
+            if importlib.util.find_spec("src")
+            else (package, local_package)
+        )
+        location = default_location
+
+        self.__load_steps_in_module(".", location)
```

Comment:
Isn't the above logic redundant? You can just try the `local_package` first and then `package` on line 28?

Reply:
The other location is tried if the default location has failed (no plugins could be loaded). 
But I guess I could improve the naming to make that a bit clearer?

Reply:
I'm saying that unless `if importlib.util.find_spec("src")` is important, you don't need `default_location`& `alternative_location` and you can just try to load `local_package` and fallback to `package` otherwise.

Reply:
Ah! I see what you mean. Yeah, this does seem double. But the problem I ran into was that `importlib.util.find_spec("src")` can return something if you use the mpyl _library_ in a local script in the mpyl repo. Or any other repo that has a `src` folder for that matter.
So it's a bit of a corner case. Ultimately the proof of the pudding is in what results in plugins being loaded. The check `find_spec` helps us look in the most likely location first.